Linphone/Views/Dialer.xaml.cs

```diff
@@ -397,38 +434,108 @@ private void EnableRegister(bool enable)
-                string loadingUrl = settingValue == null ? "http://localhost:9011" : settingValue as string;
-                Browser.Source = new Uri(loadingUrl);
-                BrowserBaseUrl = loadingUrl;
+                Browser.Source = new Uri(CallFlowControl.Instance.AgentProfile.PanelBaseUrl);
             }
         }
 
-        private void Browser_NavigationCompleted(WebView2 sender, CoreWebView2NavigationCompletedEventArgs args)
+        private async void Browser_NavigationCompleted(WebView2 sender, CoreWebView2NavigationCompletedEventArgs args)
         {
+            // HotPoint #5
             if (sender.Source.AbsolutePath == "/Account/Login")
             {
-                IsLoggedIn = false;
+                CallFlowControl.Instance.AgentProfile.IsLoggedIn = false;
                 DisableRegisteration();
             }
-            else if (sender.Source.AbsolutePath.Contains("Dashboard"))
+            else if (sender.Source.AbsolutePath.Contains("Dashboard") && CallFlowControl.Instance.AgentProfile.IsLoggedIn == false)
             {
-                IsLoggedIn = true;
+                CallFlowControl.Instance.AgentProfile.IsLoggedIn = true;
+
                 EnableRegister(true);
+
+                Browser.CoreWebView2.Navigate($"{CallFlowControl.Instance.AgentProfile.PanelBaseUrl}/api/Operators/UserInfo");
+            }
+            else if (sender.Source.AbsolutePath.Contains("/api/Operators/UserInfo"))
+            {
+                // Literally کثافتکاری
+                // Todo: Interactions with Browser should be revised.
+                try
+                {
+                    var html = await Browser.CoreWebView2.ExecuteScriptAsync("document.body.outerHTML");
+                    string content = html.Substring(html.IndexOf("sipProfile"));
+                    content = content.Substring(content.IndexOf("username"));
+                    content = content.Substring(content.IndexOf(":") + 1);
+
+                    CallFlowControl.Instance.AgentProfile.SipPhoneNumber = content.Substring(0, content.IndexOf(",")).Replace("\"", "").Replace("\\", "");
+
+                    LoadSipSettings();
+
+                    AgentStatus.IsEnabled = true;
+
+                    await AgentStatus.Dispatcher.RunIdleAsync(P =>
+                    {
+                        AgentStatus.SelectedIndex = 0;
+                    });
+                }
+                catch (Exception ex)
+                {
+                    _logger.Error(ex, "Error while reading sip phonenumber.");
+                }
+
+                Browser.CoreWebView2.Navigate($"{CallFlowControl.Instance.AgentProfile.PanelBaseUrl}");
+            }
+        }
+
+        private async void LoadSipSettings()
+        {
+            await LinphoneManager.Instance.CoreDispatcher.RunIdleAsync((args) =>
+            {
+                UpdateSettings();
+            });
+        }
+
+        private async void UpdateSettings()
```

Comment:
Why this method is needed?

Reply:
Based on the customer's request, we removed the manual configurations on the sip settings. 
By now, user configurations are stored in our databases and it get filled using this method call (First it gets settings from our core api, next it applies those settings).